components/esp_system/ld/esp32c3/sections.ld.in

```diff
@@ -220,6 +220,7 @@ SECTIONS
 
     *(.stub .gnu.warning .gnu.linkonce.literal.* .gnu.linkonce.t.*.literal .gnu.linkonce.t.*)
     *(.irom0.text) /* catch stray ICACHE_RODATA_ATTR */
+    *(.rwtext) /* catch stray .wrtext section */
```

Comment:
The comment probably has a typo: you are putting `rwtext` inside flash, yet the comment talks about `wRtext`?

Reply:
See [this](https://github.com/ivmarkov/rust-esp32-std-demo/issues/14#issuecomment-1009129358)

This patch is likely not needed after all!...